ports/zephyr/mpconfigport.h

```diff
@@ -43,11 +43,16 @@
 #define MICROPY_HELPER_REPL         (1)
 #define MICROPY_REPL_AUTO_INDENT    (1)
 #define MICROPY_KBD_EXCEPTION       (1)
-#define MICROPY_CPYTHON_COMPAT      (0)
 #define MICROPY_PY_ASYNC_AWAIT      (0)
+#ifdef CONFIG_THREAD_CUSTOM_DATA            // this one enables the _thread module
+#define MICROPY_CPYTHON_COMPAT      (1)
+#define MICROPY_PY_ATTRTUPLE        (1)
+#else
+#define MICROPY_CPYTHON_COMPAT      (0)
 #define MICROPY_PY_ATTRTUPLE        (0)
+#endif
```

Comment:
Let's just enable `MICROPY_CPYTHON_COMPAT` and `MICROPY_PY_ATTRTUPLE` (and `MICROPY_PY_BUILTINS_ENUMERATE` below) unconditionally, ie regardless of the thread setting.

And because all these are enabled by default in `py/mpconfig.h`, you can just remove their entries from this file.

(Eventually we'll move to using a `MICROPY_CONFIG_ROM_LEVEL` setting for this port, and this moves us in that direction.)

Reply:
OK. Done. Also enabled MICROPY_PY_BUILTINS_SET.